app/lib/features/space/pages/chats_page.dart

```diff
@@ -23,129 +25,141 @@ class SpaceChatsPage extends ConsumerWidget {
+                          pathParameters: {'spaceId': spaceIdOrAlias},
                         ),
-                        PopupMenuItem(
-                          onTap: () => context.pushNamed(
-                            Routes.linkChat.name,
-                            pathParameters: {'spaceId': spaceIdOrAlias},
-                          ),
-                          child: const Row(
-                            children: <Widget>[
-                              Text('Link existing Chat'),
-                              Spacer(),
-                              Icon(Atlas.chats),
-                            ],
-                          ),
+                        child: const Row(
+                          children: <Widget>[
+                            Text('Link existing Chat'),
+                            Spacer(),
+                            Icon(Atlas.chats),
+                          ],
                         ),
-                      ],
-                    ),
-                  ],
-                ),
-              );
-            },
-            orElse: () => const SliverToBoxAdapter(child: SizedBox.shrink()),
-          ),
-          chats.when(
-            data: (rooms) {
-              if (rooms.isNotEmpty) {
-                return SliverAnimatedList(
-                  initialItemCount: rooms.length,
-                  itemBuilder: (context, index, animation) => SizeTransition(
-                    sizeFactor: animation,
-                    child: ConvoCard(
-                      room: rooms[index],
-                      showParent: false,
-
-                      /// FIXME: push is broken for switching from subshell to subshell
-                      /// hence we are using `go` here.
-                      /// https://github.com/flutter/flutter/issues/125752
-                      onTap: () => context.goNamed(
-                        Routes.chatroom.name,
-                        pathParameters: {'roomId': rooms[index].getRoomIdStr()},
                       ),
-                    ),
+                    ],
+                  ),
+                ],
+              ),
+            );
+          },
+          orElse: () => const SliverToBoxAdapter(child: SizedBox.shrink()),
+        ),
+        chats.when(
+          data: (rooms) {
+            if (rooms.isNotEmpty) {
+              return SliverAnimatedList(
+                initialItemCount: rooms.length,
+                itemBuilder: (context, index, animation) => SizeTransition(
+                  sizeFactor: animation,
+                  child: ConvoCard(
+                    room: rooms[index],
+                    showParent: false,
+
+                    /// FIXME: push is broken for switching from subshell to subshell
+                    /// hence we are using `go` here.
+                    /// https://github.com/flutter/flutter/issues/125752
+                    onTap: () => isDesktop
+                        ? context.goNamed(
```

Comment:
Why ```isDesktop``` is determining ```go``` or ```push``` case ? Isn't this bug independent of platform and of core routing? 
@gnunicorn Do we agree on using ```go``` here finally, it wouldn't allow us to backtrack though.

Reply:
Added above routing condition is temporary till I finish removal of the chat-subshell in follow-up PRs. If I don't put this condition then it is miss behaving on desktop while opening chat from space.

This condition will be remove when we remove chat-subshell.